No eval.h in python 3.11

**Version(please complete the following information):**
 - Ice Version [e.g. 3.7.4]
 - OS: [e.g. Linux, Windows, macOS]
 - Distribution: [e.g. YUM, Apt, Homebrew]

**Describe the bug**
During a migration for python 3.11 on conda-forge, the following compilation error was encountered:

```
src/Slice.cpp:17:10: fatal error: eval.h: No such file or directory
     17 | #include <eval.h>
        |          ^~~~~~~~
```
This is due to the following change:

> The non-limited API files cellobject.h, classobject.h, code.h, context.h, funcobject.h, genobject.h and longintrepr.h have been moved to the Include/cpython directory. Moreover, the eval.h header file was removed. These files must not be included directly, as they are already included in Python.h: [Include Files](https://docs.python.org/3/c-api/intro.html#api-includes). If they have been included directly, consider including Python.h instead.

**Expected behavior**
It should compile

Xref: https://github.com/conda-forge/zeroc-ice-feedstock/pull/30

--- a/python/modules/IcePy/Slice.cpp
+++ b/python/modules/IcePy/Slice.cpp
@@ -14,7 +14,12 @@
 // Python headers needed for PyEval_EvalCode.
 //
 #include <compile.h>
-#include <eval.h>
+// Use ceval.h instead of eval.h with Pyhthon 3.11 and greater
+#if PY_VERSION_HEX > 0x030B0000
+#   include <ceval.h>
+#else
+#   include <eval.h>
+#endif
 
 using namespace std;
 using namespace IcePy;

--- a/python/modules/IcePy/Slice.cpp
+++ b/python/modules/IcePy/Slice.cpp
@@ -15,7 +15,7 @@
 //
 #include <compile.h>
 // Use ceval.h instead of eval.h with Pyhthon 3.11 and greater
-#if PY_VERSION_HEX > 0x030B0000
+#if PY_VERSION_HEX >= 0x030B0000
 #   include <ceval.h>
 #else
 #   include <eval.h>

--- a/python/modules/IcePy/Util.cpp
+++ b/python/modules/IcePy/Util.cpp
@@ -225,7 +225,12 @@ IcePy::getFunction()
     //
     // Get name of current function.
     //
+    // Use PyEval_GetFrame with Pyhthon >= 3.11
+#if PY_VERSION_HEX > 0x030B0000
+    PyFrameObject *f = PyEval_GetFrame();
+#else
     PyFrameObject *f = PyThreadState_GET()->frame;
+#endif
     PyObjectHandle code = getAttr(reinterpret_cast<PyObject*>(f), "f_code", false);
     assert(code.get());
     PyObjectHandle func = getAttr(code.get(), "co_name", false);

--- a/python/modules/IcePy/Util.cpp
+++ b/python/modules/IcePy/Util.cpp
@@ -226,7 +226,7 @@ IcePy::getFunction()
     // Get name of current function.
     //
     // Use PyEval_GetFrame with Pyhthon >= 3.11
-#if PY_VERSION_HEX > 0x030B0000
+#if PY_VERSION_HEX >= 0x030B0000
     PyFrameObject *f = PyEval_GetFrame();
 #else
     PyFrameObject *f = PyThreadState_GET()->frame;

